src/Tools/dotnet-monitor/LibrarySharing/DefaultSharedLibraryInitializer.cs

```diff
@@ -86,6 +152,22 @@ public IFileProviderFactory Initialize()
             return new Factory(sharedLibraryPath);
         }
 
+        private static int ComputeHashAndLength(SafeFileHandle handle, Span<byte> destination)
+        {
+            int length = (int)RandomAccess.GetLength(handle);
```

Comment:
Consider separating the length and hash checks. If the lengths don't match, there is no need to compute the hash.